lib/type.js

```diff
@@ -33,14 +21,15 @@ var natives = {
  * @return {String} object type
  * @api public
  */
+var objectTypeRegexp = /^\[object (.*)\]$/;
 
 function getType(obj) {
-  var str = Object.prototype.toString.call(obj);
-  if (natives[str]) return natives[str];
-  if (obj === null) return 'null';
-  if (obj === undefined) return 'undefined';
-  if (obj === Object(obj)) return 'object';
-  return typeof obj;
+  var type = Object.prototype.toString.call(obj).match(objectTypeRegexp)[1].toLowerCase();
+  if (type === 'string' && typeof obj === 'object') return 'object'; // Let "new String('')" return 'object'
```

Comment:
ocd: Can you put the comments before the if statement. Makes for more logical reading and helps avoid line length greater than 80.